cache/remote.go

```diff
@@ -102,30 +103,38 @@ func (sr *immutableRef) GetRemote(ctx context.Context, createIfNeeded bool, comp
 					existingRepos = append(existingRepos, repo)
 				}
 				desc.Annotations[dslKey] = strings.Join(existingRepos, ",")
+				if dh, ok := sr.descHandlers[desc.Digest]; ok {
+					desc.Annotations = mergeEStargzAnnotations(dh.SnapshotLabels, desc.Annotations)
```

Comment:
Just a note. It would be better if this annotations load/save wouldn't need startgz-only exceptions everywhere and would be more generic. But I don't have a specific proposal how to improve this.

Reply:
Thank you for the note. Fixed the patch to make it more generic. Now `*immutableRef.addCompressionBlob()` saves estargz-related annotations as well so we don't need estargz-only expections.